### PR TITLE
Contributing and conventions

### PR DESCRIPTION
**Pull request by freaz**
## Description

This improves contributing guides and adds conventions.
Also `commmunication/send-email` was updated to follow the conventions except One usecase per Profile. I will do that as separate PR, as it is bigger change not entirely needed for conventions.

## Motivation and Context

We want to have capabilities in great state and set role model for others how they could be written.

Focused mainly on [CONVENTIONS.md](https://github.com/superfaceai/station/pull/62/files#diff-2e28754e69aa417e5b6d89e99e42f05bfb6330800fa823753383db1d170fbc2f). There is more work to be done to make the repository play nice with all the tooling.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have updated the documentation accordingly. For updating Oclif commands documentation use [oclif-dev](https://github.com/oclif/dev-cli#oclif-dev-readme).
- [x] I have read the **CONTRIBUTION_GUIDE** document.
- [ ] I haven't repeated the code. (DRY)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: CONVENTIONS.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Convetions
+
+This document gives coding conventions for the Superface capabilities created in [Station repository](https://github.com/superfaceai/station).
+
+## Profile
+
+### Structure
+
+- Use 2 spaces per indentation level.
+- Profile should contain one usecase.
```
Comment (zdne) [inline CONVENTIONS.md:10]: Maybe let's follow https://datatracker.ietf.org/doc/html/rfc2119 and use all capitals: 

> - Profile SHOULD contain one usecase.
> - Usecase MUST define safety.

File: CONVENTIONS.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+### Descriptions
+
+- Use block string literal `"""` when writing title and description.
+- Profile must have title and description.
+- Every usecase must have title and description.
+- Every field must have title and description.
+
+### Map
+
+We need your feedback.
```
Comment (zdne) [inline CONVENTIONS.md:31]: What this mean? Feedback on maps or feedback on the conventions for map?

File: CONVENTIONS.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+### Provider
+
+- Provider name must use `-` (dash) as delimiter.
+- Provider Id should be short and descriptive.
```
Comment (zdne) [inline CONVENTIONS.md:36]: maybe add:

> Provider id MUST NOT be misleading

?

File: CONVENTIONS.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+- Default service Id for versioned APIs should match the version.
+- Security Scheme Id must use `-` (dash) as delimiter.
+- Security Scheme Id should be named same as in provider documentation.
+
+### Tests
+
+- Capability must have automated tests.
+- Test must use network recording.
+- Test must use snapshot for asserting.
```
Comment (zdne) [inline CONVENTIONS.md:45]: Maybe add "TODO" and link to @martinalbert testing set when available

Comment (zdne) [reply CONVENTIONS.md:45]: Finally, maybe add a link to a role model profile?

**Review by zdne — COMMENTED** [review]
Looks good. Some minor comments

File: CONVENTIONS.md (hunk 4 above)
Comment (freaz) [reply CONVENTIONS.md:45]: I planned to add how to write tests in the README and update existing. Until it is ready, I will replace last two point with todo to add link to readme how tests should be done.

Ok I will link it

**Comment by freaz** [comment]
Thanks for the comments. Hopefully worked all of them in.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -22,20 +22,24 @@ If you have any question about this project (for example, how to use it) or if y
...
 
 **NOTE: Be sure to merge the latest from "upstream" before making a pull request.**
 
 **NOTE: Please open an issue first if you want to make larger changes**
 
+### Contribute by adding capability
+
+For help with authoring new capability see [Superface documentation](https://superface.ai/docs/guides/create-new-capability).
+
+In Station we want to keep capabilities which are consistent abd in great quality. To achieve it, set of [convetions](CONVENTIONS) were created.
```
Comment (kysely) [inline CONTRIBUTING.md:41]:
```suggestion
In Station we want to keep capabilities which are consistent and in great quality. To achieve it, set of [convetions](CONVENTIONS) were created.
```

File: CONVENTIONS.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
+# Convetions
```
Comment (kysely) [inline CONVENTIONS.md:1]:
```suggestion
# Conventions
```

File: CONVENTIONS.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+- Use block string literal `"""` when writing title and description.
+- Profile MUST have title and description.
+- Every usecase MUST have title and description.
+- Every field MUST have title and description.
+
+### Map
+
+There are no particular conventions right now.
+We want first gather more content to understand how Complink maps are written to decide what conventions would make sense.
```
Comment (kysely) [inline CONVENTIONS.md:32]:
```suggestion
We want first gather more content to understand how Comlink maps are written to decide what conventions would make sense.
```

File: CONVENTIONS.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+### Provider
+
+- Provider name MUST use `-` (dash) as delimiter.
+- Provider Id SHOULD be short and descriptive.
+- Provider id MUST NOT be misleading.
```
Comment (kysely) [inline CONVENTIONS.md:38]: Consistent case for `Id`

```suggestion
- Provider Id MUST NOT be misleading.
```

File: capabilities/communication/send-email/profile.supr [hunk 9]
```diff
@@ -28,8 +27,8 @@ usecase SendEmail unsafe {
 }
 
 """
-Send templated transactional email to one recipient
-
+Send Teamplated Email
```
Comment (kysely) [inline capabilities/communication/send-email/profile.supr:30]:
```suggestion
Send Templated Email
```

File: capabilities/communication/send-email/profile.supr [hunk 10]
```diff
@@ -48,42 +47,64 @@ usecase SendTemplatedEmail unsafe {
...
 field templateId
 
-"Template Data
-  Template data to be applied to the specified template to generate html, test, and subject.
-  This field is a collection of key/value pairs following the pattern `variable_name`: `value to insert`.
-  This field should be used in combination with `templateId` to identify what template to use."
+"""
+Template Data
+Template data to be applied to the specified template to generate html, test, and subject.
+his field is a collection of key/value pairs following the pattern `variable_name`: `value to insert`.
```
Comment (kysely) [inline capabilities/communication/send-email/profile.supr:101]:
```suggestion
This field is a collection of key/value pairs following the pattern `variable_name`: `value to insert`.
```

**Review by kysely — COMMENTED** [review]
👏 Great work on this.

Just found few typos when reading this :)

**Comment by freaz** [comment]
@kysely Thanks a lot, selective blindness works well for me.

**Pull request merged**